src/test/java/mate/academy/spring/boot/controllerTests/BookControllerTest.java

```diff
@@ -0,0 +1,289 @@
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/add-books-to-books-table.sql"));
+        }
+    }
+
+    @AfterEach
+    void afterAll(
+            @Autowired DataSource dataSource
+    ) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/remove-books-from-books-table.sql")
+            );
+        }
+    }
+
+    @Test
+    @WithMockUser
+    @DisplayName("Should return List BookDtoS")
+    public void testGetAllBooks_shouldReturnListOfBooksDto() throws Exception{
+        List<BookDto> expected = new ArrayList<>();
+        BookDto firstBook = getBookDto("Book 1", "Author 1", BigDecimal.valueOf(20.00));
+        expected.add(firstBook);
+        BookDto secondBook = getBookDto("Book 2", "Author 2", BigDecimal.valueOf(33.00));
+        expected.add(secondBook);
+        MvcResult result = mockMvc.perform(get("/books"))
+                .andExpect(status().isOk())
+                .andReturn();
+        List<BookDto> actual = objectMapper.readValue(result.getResponse().getContentAsString(),
+                objectMapper.getTypeFactory().constructCollectionType(List.class, BookDto.class));
+        Assertions.assertNotNull(actual);
+        Assertions.assertFalse(actual.isEmpty());
+        EqualsBuilder.reflectionEquals(expected, actual, "id");
+    }
+
+    @Test
+    @DisplayName("Should return book by ID")
+    @WithMockUser
+    public void testGetFindById_withCorrectId_shouldReturnBook() throws Exception {
+        BookDto expected = getBookDto("Book 1", "Author 1", BigDecimal.valueOf(20.00));
+        Long bookId = 1L;
+
+        MvcResult result = mockMvc.perform(get("/books/{id}", bookId))
+                .andExpect(status().isOk())
+                .andReturn();
+        BookDto actual = objectMapper.readValue(result.getResponse().getContentAsString(), BookDto.class);
+        System.out.println();
+        Assertions.assertNotNull(actual);
+        Assertions.assertNotNull(actual.getId());
+        EqualsBuilder.reflectionEquals(expected, actual, "id");
+    }
+
+    @Test
+    @WithMockUser
+    @DisplayName("Should return EntityNotFoundException")
+    public void testGetBookById_withIncorrectId_ShouldReturnNotFound() throws Exception {
+        Long invalidBookId = 999L;
+        mockMvc.perform(get("/books/{id}", invalidBookId))
+                .andExpect(status().isNotFound());
+    }
+
+    @Test
+    @DisplayName("Should search books by title")
+    @WithMockUser
+    public void getBooksByTitle_withCorrectTitle_shouldReturnListOfBooks() throws Exception {
```

Comment:
I noticed you have some test methods that start with "test" and some that don't. Please, let's follow the same naming convention for all methods and remove "test" part, like here